### PR TITLE
Minor changes allowing reaction support

### PR DESCRIPTION
**Pull request by JJ-Atkinson**

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -241,15 +241,18 @@ export const Channel = {
   messages: () => ({}),
   sendMessage: async (args, { self }) => {
     const { id } = self.$argsAt(root.guilds.one.channels.one);
+    const msg = {
+      content: args.content,
+      components: JSON.parse(args.components || "[]"),
+      embeds: JSON.parse(args.embeds || "[]"),
+    };
+    if (args.message_reference) msg["message_reference"] = args.message_reference;
+    if (args.allowed_mentions) msg["allowed_mentions"] = args.allowed_mentions;
```
Comment (JJ-Atkinson) [inline index.ts:250]: Add support for message reference (i.e. "reply to") and allowed mentions configuration.

File: index.ts [hunk 2]
```diff
@@ -264,6 +267,26 @@ export const Message = {
...
+  reactionsByEmote: async (args, { self }) => {
+    const { id: channelId } = self.$argsAt(root.guilds.one.channels.one);
+    const { id: messageId } = self.$argsAt(root.guilds.one.channels.one.messages.one);
+    const emoteStr = args.emote_str;
+    const res = await api(
+      "GET",
+      `channels/${channelId}/messages/${messageId}/reactions/${emoteStr}`,
+      null);
+    return await res.json();
+  },
```
Comment (JJ-Atkinson) [inline index.ts:279]: Returns the reactions per emote on a message

File: index.ts [hunk 3]
```diff
@@ -264,6 +267,26 @@ export const Message = {
...
+  postReaction: async (args, { self }) => {
+    const { id: channelId } = self.$argsAt(root.guilds.one.channels.one);
+    const { id: messageId } = self.$argsAt(root.guilds.one.channels.one.messages.one);
+    const emoteStr = args.emote_str;
+    const res = await api(
+      "PUT",
+      `channels/${channelId}/messages/${messageId}/reactions/${emoteStr}/@me`,
+      null
+    );
+  }
```
Comment (JJ-Atkinson) [inline index.ts:289]: Allows posting of a reaction on a message

File: memconfig.json [hunk 4]
```diff
@@ -486,7 +486,7 @@
         "actions": [
           {
             "name": "sendMessage",
-            "type": "Void",
+            "type": "Json",
```
Comment (JJ-Atkinson) [inline memconfig.json:489]: Allow retrieval of the message id of the sent message.

**Review by juancampa — APPROVED** [review]
Looks great. Thanks!

**Pull request merged**